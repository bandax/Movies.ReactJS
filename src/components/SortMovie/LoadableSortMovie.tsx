import * as React from 'react';
import Loadable from 'react-loadable';
import Loading from '../Loading/Loading';
import { ISortOption } from '../../interfaces/ISortMovie';

const LoadableSortMovie = Loadable({
  loader: () => import('./SortMovie'),
  loading: Loading,
});

interface ISortMovieProps {
  sortOptions: ISortOption[];
}

const SortMovieSplit: React.FunctionComponent<ISortMovieProps> = (
  props: ISortMovieProps,
) => {
  return <LoadableSortMovie sortOptions={props.sortOptions} />;
};

export default SortMovieSplit;
