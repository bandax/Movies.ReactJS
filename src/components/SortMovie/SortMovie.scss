.sort-title {
  text-transform: uppercase;
}
