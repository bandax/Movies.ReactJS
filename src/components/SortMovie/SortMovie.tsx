import * as React from 'react';
import './SortMovie.scss';
import { ISortOption } from '../../interfaces/ISortMovie';

interface ISortMovieProps {
  sortOptions: ISortOption[];
}

const SortMovie: React.FunctionComponent<ISortMovieProps> = (
  props: ISortMovieProps
) => {
  const { sortOptions } = props;

  return (
    <>
      <span className="sort-title">sort by: </span>
      <select>
        {sortOptions.map((sortOption: ISortOption) => (
          <option key={sortOption.id} value={sortOption.id}>
            {sortOption.name}
          </option>
        ))}
      </select>
    </>
  );
};

export default SortMovie;
