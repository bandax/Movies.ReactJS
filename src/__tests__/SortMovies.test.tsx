import * as React from 'react';
import { SortMovie } from '../components/SortMovie/SortMovie';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import sortOptions from '../data/sorts.json';

describe('<Footer />', () => {
  it('render a Footer Component', async () => {
    const { asFragment } = render(<SortMovie sortOptions={sortOptions} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
