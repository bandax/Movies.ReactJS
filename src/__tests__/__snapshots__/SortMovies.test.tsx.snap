// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> render a Footer Component 1`] = `
<DocumentFragment>
  <span
    class="sort-title"
  >
    sort by: 
  </span>
  <select>
    <option
      value="sort-1"
    >
      Release Date
    </option>
    <option
      value="sort-2"
    >
      Most Popular
    </option>
  </select>
</DocumentFragment>
`;
